### DISCUSSION TITLE
modular composition ? 🤷🏿‍♂️

### DISCUSSION THREAD

**Discussion by ceddybi** [hayes/pothos#1330] [Ideas]
Is it possible to publish packages of resolvers then combine them in the final app?

With something like type-graphql, i can create resolvers and publish them as separate npm packages, then if i want to use them i simply download them and combine them from multiple packages to create a final schema, 

How can i achieve a similar approach with POTHOS?

**Comment by ceddybi** [comment]
@hayes

**Comment by hayes** [comment]
The very generic solution is the add-graphql plugin which let's you merge in any graphql types from another schema

I haven't tested it out yet, but the internal architecture of Pothos 4.0 was redesigned to allow for referencing types defined by other builder instances.



Have you tried just defining types in a separate builder and seeing if it works? (I won't be surprised if there are some issues)

If you are doing this with packages, I'd use something like yarn/pnpm resolutions to ensure you only have one copy of Pothos (along with any other plugins)
